lib/pbench/test/unit/server/test_datasets_compare.py

```diff
@@ -140,16 +140,20 @@ def mock_extract(_tarball_path: Path, _path: str):
                 HTTPStatus.UNSUPPORTED_MEDIA_TYPE,
                 "Unsupported Benchmark: hammerDB",
             ),
-            # Add more scenarios here.
         ),
     )
     def test_datasets_with_different_benchmark(
         self, user, datasets, exp_status, exp_message, query_get_as, monkeypatch
     ):
+        def mock_compare_csv_to_json(
+            self, benchmark_name, input_type, data_stream
+        ) -> JSON:
+            return {"status": "success", "json_data": "quisby_data"}
```

Comment:
Not really that important: but, for the record, this unconditional mock means your "if success" asserts below aren't doing anything beyond verifying that `monkeypatch` worked; only the assert in the `else` is actually interesting. But there are worse things than redundant checks. 😆

Reply:
I don't _think_ that's true (although, my track record on thinking today is not good...):  this test is parametrized, and there are ways other than this mock to generate failures....  (And, the cases where `compare_csv_to_json()` generates failures are tested in tests other than this one.)

Reply:
> this test is parametrized, and there are ways other than this mock to generate failures.... (And, the cases where `compare_csv_to_json()` generates failures are tested in tests other than this one.)

I'm confused, but I don't think it matters. The `if` case I'm talking about means we succeeded on the strength of this mock, because I don't think there's any other path to success given the setup. The failure cases here go to the `else`, and failure cases elsewhere have nothing to do with this mock or this test case since they're isolated.

More importantly, perhaps, I made comments but approved. In fact I think I'll just resolve these comments. Are we ready to merge this? 😆

Reply:
> The `if` case I'm talking about means we succeeded on the strength of this mock, because I don't think there's any other path to success given the setup.

Correct; but, with different parametrization, we won't get to this mock, and so we need to use different asserts in those cases.

> The failure cases here go to the `else`, and failure cases elsewhere have nothing to do with this mock or this test case since they're isolated.

Agreed; but, since this test tests both failure and success cases, we need the `if` statement.  And, in the success case, those assertions are checking that the value returned by the mock are subsequently returned by the API, so I think they are more than just redundant.  (And, I don't think those scenarios are checked anywhere else.)

> Are we ready to merge this? 😆

I'm waiting to see the output of the tests, which are almost done, now.